packages/compass-indexes/src/stores/store.ts

```diff
@@ -0,0 +1,116 @@
+import { createStore, applyMiddleware } from 'redux';
+import reducer from '../modules';
+import thunk from 'redux-thunk';
+import {
+  localAppRegistryActivated,
+  globalAppRegistryActivated,
+} from '@mongodb-js/mongodb-redux-common/app-registry';
+import { writeStateChanged } from '../modules/is-writable';
+import { getDescription } from '../modules/description';
+import {
+  fetchIndexes,
+  inProgressIndexAdded,
+  inProgressIndexRemoved,
+  inProgressIndexFailed,
+  type InProgressIndex,
+} from '../modules/regular-indexes';
+import { SearchIndexesStatuses } from '../modules/search-indexes';
+import type { DataService } from 'mongodb-data-service';
+import type AppRegistry from 'hadron-app-registry';
+
+export type IndexesDataService = Pick<
+  DataService,
+  'indexes' | 'isConnected' | 'updateCollection' | 'createIndex' | 'dropIndex'
+>;
+export type ConfigureStoreOptions = {
+  dataProvider: {
+    dataProvider?: IndexesDataService;
+  };
+  namespace: string;
+  localAppRegistry: AppRegistry;
+  globalAppRegistry: AppRegistry;
+  serverVersion: string;
+  isReadonly: boolean;
+  isSearchIndexesSupported: boolean;
+};
+
+const configureStore = (options: ConfigureStoreOptions) => {
+  if (!options.dataProvider?.dataProvider) {
+    throw new Error(
+      "Can't configure store for indexes plugin without data serivce"
+    );
+  }
+  const store = createStore(
+    reducer,
+    {
+      dataService: options.dataProvider.dataProvider,
+      namespace: options.namespace,
+      serverVersion: options.serverVersion,
+      isReadonlyView: options.isReadonly,
+      searchIndexes: {
+        status: options.isSearchIndexesSupported
+          ? SearchIndexesStatuses.PENDING
+          : SearchIndexesStatuses.NOT_AVAILABLE,
+      },
+    },
+    applyMiddleware(thunk.withExtraArgument({}))
+  );
+
+  // Set the app registry if preset. This must happen first.
+  if (options.localAppRegistry) {
+    const localAppRegistry = options.localAppRegistry;
+    store.dispatch(localAppRegistryActivated(localAppRegistry));
```

Comment:
If those are not in the store, we don't need to dispatch this, same for global one

Reply:
we probably need it because we trigger delete/create modals using localAppRegistry. And by removing this, it not able to find the registry and throws.

Reply:
Yes, drop index and create index stores both have app registry slice, but this is a completely different store, you literally don't have a slice here that would react to those actions

Reply:
Wait i missed adding the registry. We have a default slice from the redux-common which is used in this store and then used on the UI to trigger actions.

I added this commit to the parent branch [91195c6](https://github.com/mongodb-js/compass/pull/4815/commits/91195c6056e6fe0951a4b483ed046b98c1d13482)